src/Display.java

```diff
@@ -0,0 +1,43 @@
+public class Display {
+
+    public static void showDataAnimal(GenericContainer<? extends Animal> container) {
+        System.out.println(container.getT());
+        System.out.println(container.getAnimal());
+        System.out.println(container.getMammal());
```

Comment:
this here is not necessary -> the Container class should only hold a reference to T -> no explicitly created Objects. It is supposed to be typically GENERIC - and if you write the objects inside it, the meaning of the generics is a bit pointless